### ISSUE TITLE
CI tests are not catching issues with observatorium-api and cortex

### ISSUE THREAD

**Issue by nmagnezi** [observatorium/configuration#211]
Related Issue https://github.com/observatorium/configuration/issues/208

**Comment by rollandf** [comment]
Here are `up` logs when running with the wrong Cortex image:
```
name=up caller=level.go:63 ts=2020-03-19T08:54:28.412375559Z level=info caller=main.go:164 msg="caught interrupt"
name=up caller=level.go:63 ts=2020-03-19T08:54:28.412676574Z level=info caller=main.go:403 component=reader msg="number of requests" success=0 errors=113
name=up caller=level.go:63 ts=2020-03-19T08:54:28.41270087Z level=error caller=main.go:407 component=reader msg="ratio is below threshold"
name=up caller=level.go:63 ts=2020-03-19T08:54:28.41271735Z level=info caller=main.go:256 msg="up completed its mission!"
```
And the `e2e.sh` output is: `job.batch/observatorium-up condition met`

**Comment by brancz** [comment]
that should indeed be fixed, this should have failed

**Comment by rollandf** [comment]
It seems the problem is in `up` code.
`up` will exit with error if the `run group` finishes with error:
https://github.com/observatorium/up/blob/master/main.go#L262

However, the `run group` will return the error returned by the first exiting actor:
https://github.com/oklog/run/blob/master/group.go#L28

In general, the writer will finish first.
It can be reproduce in `up` repo by running `make test-integration` with a wrong URL.
Changing `endpoint-read` will not fail, and changing `endpoint-write` will fail.

**Comment by brancz** [comment]
Makes sense. Do you want to fix that @rollandf ? We should only exit the writer run group when done no? Maybe put them in the same run group to make that coordination easier?

cc @squat @kakkoyun

**Comment by squat** [comment]
nice find there :+1:

**Comment by rollandf** [comment]
I can try to fix it.
There is currently only one run group. Should I split the writer and the reader in different groups?

**Comment by brancz** [comment]
No strong opinion. Happy with any strategy that would fix it :)

**Comment by rollandf** [comment]
Proposed fix here: https://github.com/observatorium/up/pull/12

**Comment by rollandf** [comment]
Fixed by #230
